src/lib/PriorityFeeLib.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {OutputToken, InputToken} from "../base/ReactorStructs.sol";
+import {PriorityOutput, PriorityInput} from "../lib/PriorityOrderLib.sol";
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+
+/// @notice helpers for handling priority order objects
+library PriorityFeeLib {
+    using FixedPointMathLib for uint256;
+
+    /// @notice we denominate priority fees in terms of micro-points, or 1 millionith of a basis point
+    uint256 constant MPS = 1e10;
+
+    /// @notice returns a scaled input using the current priority fee and mpsPerPriorityFeeWei
+    /// @notice the amount is scaled down to favor the swapper
+    /// @param input the input to scale
+    /// @param priorityFee the current priority fee
+    /// @return a scaled input
+    function scale(PriorityInput memory input, uint256 priorityFee) internal pure returns (InputToken memory) {
+        if (priorityFee * input.mpsPerPriorityFeeWei >= MPS) {
+            return InputToken({token: input.token, amount: 0, maxAmount: 0});
```

Comment:
ok crazy but is there ever a case where the MEV is so high they'll pay YOU?

Reply:
discussed offline - don't think this can happen since the MEV is naturally capped at the max input offered by the swapper